SonarQube.TeamBuild.PreProcessor/ArgumentProcessor.cs

```diff
@@ -184,113 +114,17 @@ public static ProcessedArgs TryProcessArgs(string[] commandLineArgs, ILogger log
-                
-                if(TryGetMatchingDescriptor(arg, out descriptor, out prefix))
-                {
-                    KeywordIds newId = descriptor.Id;
-
-                    if (!descriptor.AllowMultiple && IdExists(newId, arguments))
-                    {
-                        string existingValue = TryGetArgumentValue(newId, arguments);
-                        logger.LogError(Resources.ERROR_CmdLine_DuplicateArg, arg, existingValue);
-                        parsedOk = false;
-                    }
-                    else
-                    {
-                        // Store the argument
-                        string argValue = arg.Substring(prefix.Length);
-                        arguments.Add(new ArgumentInstance(descriptor, argValue));
-                    }
-                }
-                else
-                {
-                    logger.LogError(Resources.ERROR_CmdLine_UnrecognisedArg, arg);
-                    parsedOk = false;
-                }
-            }
-
-            argumentInstances = arguments;
-            return parsedOk;
-        }
-
-        private static bool TryGetMatchingDescriptor(string argument, out ArgumentDescriptor descriptor, out string prefix)
-        {
-            descriptor = null;
-            prefix = null;
-
-            bool found = false;
-
-            foreach(ArgumentDescriptor item in Descriptors)
-            {
-                string match = TryGetMatchingPrefix(item, argument);
-                if (match != null)
-                {
-                    descriptor = item;
-                    prefix = match;
-                    found = true;
-                    break;
-                }
-            }
-            return found;
-        }
-
-        private static string TryGetMatchingPrefix(ArgumentDescriptor descriptor, string argument)
-        {
-            Debug.Assert(descriptor.Prefixes.Where(p => argument.StartsWith(p, StringComparison.OrdinalIgnoreCase)).Count() < 2,
-                "Not expecting the argument to match multiple prefixes");
-
-            string match = descriptor.Prefixes.FirstOrDefault(p => argument.StartsWith(p, StringComparison.OrdinalIgnoreCase));
-            return match;
-        }
-
-        private static bool IdExists(KeywordIds id, IEnumerable<ArgumentInstance> arguments)
-        {
-            bool exists = arguments.Any(a => a.Descriptor.Id == id);
-            return exists;
-        }
-
-        private static ArgumentInstance TryGetArgument(KeywordIds id, IEnumerable<ArgumentInstance> arguments)
+        private static ArgumentInstance TryGetArgument(string id, IEnumerable<ArgumentInstance> arguments)
         {
             return arguments.FirstOrDefault(a => a.Descriptor.Id == id);
         }
 
-        private static string TryGetArgumentValue(KeywordIds id, IEnumerable<ArgumentInstance> arguments)
+        private static string TryGetArgumentValue(string id, IEnumerable<ArgumentInstance> arguments)
```

Comment:
The comparison in this method should specify a string comparer too. I'll change it.